cpp/src/arrow/python/arrow_to_pandas.h

```diff
@@ -64,6 +64,12 @@ struct PandasOptions {
   /// objects
   bool deduplicate_objects = false;
 
+  /// \brief For certain data types, a cast is needed in order to store the
+  /// data in a pandas DataFrame or Series (e.g. timestamps are always stored
+  /// as nanoseconds in pandas). This option controls whether it is a safe
+  /// cast or not.
+  bool safe = true;
```

Comment:
Should we rename this to `safe_cast` (or `checked_cast`) for clarity perhaps?

Reply:
In the C++ API that's fine, but in the Python API it might be more consistent to use `safe`, since `Table.from_pandas()` uses the same keyword (for in case it needs to cast to follow a specified schema). Although I was having the exact same question in my head while writing this, as `safe` doesn't say much on itself.

Reply:
Shall I change the name in C++ ?

Reply:
Yes, I think `safe_cast` or `checked_cast` would be better.

Reply:
OK, renamed to `safe_cast` on the C++ side